private/server/worker/tree.ts

```diff
@@ -342,17 +343,23 @@ const getRenderingTree = (
 ) => {
   let element: ReactNode = null;
   let components: Record<string, React.ComponentType<unknown>> = {};
+  let toc: Toc = [];
 
   for (const [path, leaf] of leaves.entries()) {
+    toc = toc.concat(leaf.tableOfContents || []);
     if (!path.includes('layout.')) continue;
     components = { ...components, ...leaf.components };
   }
 
   for (const [, leaf] of leaves.entries()) {
     if (element) {
-      element = React.createElement(leaf.default, { components }, element);
+      element = React.createElement(
+        leaf.default,
+        { components, tableOfContents: toc },
+        element,
+      );
     } else {
-      element = React.createElement(leaf.default, { components });
+      element = React.createElement(leaf.default, { components, tableOfContents: toc });
```

Comment:
Does this mean we're passing an array of TOCs to layouts? Why is that?

I mean, MDX pages can't be nested, so I think there can only ever be one table of contents?